### ISSUE TITLE
Compilation problems

### ISSUE THREAD

**Issue by Alysko** [papampi/nvOC_miners#7]
Hi, the update script checks version and doesn't clone git if version match.
Ok, it's a little confusion but why not.

But there is a problem: if - it's what happens to me - binaries cloned from git doesn't work, we cannot compile from source because the update script considers that version is ok, so doesn't download source submodule.

Moreover, some sources doesn't have the good files permissions and compilation aborts. The best I think is to add a:
`chmod +x ${NVOC_MINERS}/*/src/*.sh`
or something similar.

**Comment by LuKePicci** [comment]
Updated sources will be checked out if needed only when you ask to compile them. If you don’t compile anything no sources are cloned. Version checking is only intended for binaries. Sources are versioned by git submodules commit hash. Binary version checking has nothing to dom and does not prevent you to start compilation. As soon as you start ask for compilation the submodule should be eventually updated/initialized. Again, in no case a binary version check resulting in “already up to date” will casue you to not have updated sources every time you compile them starting from the nvOC update script.

________________________________
Da: BenM <notifications@github.com>
Inviato: Wednesday, June 6, 2018 2:44:40 PM
A: papampi/nvOC_miners
Cc: Subscribed
Oggetto: [papampi/nvOC_miners] Compilation problems (#7)


Hi, the update script checks version and doesn't clone git if version match.
Ok, it's a little confusion but why not.

But there is a problem: if - it's what happens to me - binaries cloned from git doesn't work, we cannot compile from source because the update script considers that version is ok, so doesn't download source submodule.

Moreover, some sources doesn't have the good files permissions and compilation aborts. The best I think is to add a:
chmod +x ${NVOC_MINERS}/*/src/*.sh
or something similar.

**Comment by Alysko** [comment]
Ok, so there is a problem because I could not compile ANXccminer and others: src was empty.

And for some miners, .sh files is not executable (-> chmod +x as I suggested previously)

**Comment by LuKePicci** [comment]
Please confirm that running `bash build.sh` works the same as making the shell script executable and directly running it.

**Comment by Alysko** [comment]
I confirm.

**Comment by LuKePicci** [comment]
The git submudule update command for ANX ccminer is here:
https://github.com/papampi/nvOC_miners/blob/91146487ceadc807d5853e559020042e791f2e22/nvOC_miner_update.sh#L352
What happens when you run the nvOC update script from miners folder and you ask to compile ANX?

**Comment by Alysko** [comment]
It's what I done manually.
When I run update script, it said that it's already compile and there's nothing in /src.

**Comment by LuKePicci** [comment]
But there is no kind of "already compiled" check. That command is always  executed when you choose option 9...

**Comment by Alysko** [comment]
Yes, you're right. But I'm pretty sure to have start a ANXccminer which doesn't work for that reason... Now I change the update script to work (chmod +x) and launch submodule update manually.

**Comment by LuKePicci** [comment]
Ok I just tried cloning the miners repo from scratch. When you choose option 9 it will fail to launch because of problems with paths. Will fix it.

**Comment by Alysko** [comment]
And add the chmod +x or bash to allow compilation! Thanks.

**Comment by LuKePicci** [comment]
Just committed, check if it's working as expected now.
